module/VuFindSearch/src/VuFindSearch/Backend/WorldCat/Response/XML/RecordCollectionFactory.php

```diff
@@ -67,11 +66,7 @@ class RecordCollectionFactory implements RecordCollectionFactoryInterface
      */
     public function __construct($recordFactory = null, $collectionClass = null)
     {
-        if (null === $recordFactory) {
-            $recordFactory = function ($i) {
```

Comment:
I'm not sure I understand why this needed to be removed. I think the intention here was to have a simple container class for fetching WorldCat records when a full record driver was not needed. It's probably unused, but I'm not sure there's a compelling reason to get rid of it. If we do want to get rid of it, it might also be better to actually delete the Record class rather than leaving it here unused... but I'd be inclined to keep things as they are just to minimize impact at this time (unless I'm overlooking a strong reason that the removal is needed).

Reply:
It didn't need to be removed, but since it was unused and needed to be changed I thought it could just as well be dropped, but I'll drop the changes from here. In the long run we may want to revisit at least EDS, EIT and LibGuides as well.

Reply:
Would it be worth deprecating the unused classes in 8.1 and removing them in 9.0, perhaps? I doubt anyone is going to care about them, but that at least gives some fair warning.

Reply:
That would make sense, though if there's really a use case for them, I have no problem keeping them around.